File: experiments/020-PythonVsJava/cells/java-gradle_1-Empty_free/2019-2c-parcial2-masked/20260921-202808-59615/workdir/src/main/java/surveycrawler/FirmSand.java
package surveycrawler;

public class FirmSand extends GroundType {

    // moving

    @Override
    public void moveCrawlerInDirection(SurveyCrawler aCrawler, Point aDirection) {
        aCrawler.moveBy(aDirection);
    }

    // facing

    @Override
    public void turnCrawler(SurveyCrawler aCrawler, Runnable aTurn) {
        aTurn.run();
    }
}

File: experiments/020-PythonVsJava/cells/python-pytest_1-Empty_free/2019-2c-parcial2-masked/20260921-212215-18237/analysis/acceptance-project/src/surveycrawler/firm_sand.py
from surveycrawler.ground_type import GroundType


class FirmSand(GroundType):

    # moving

    def move_crawler_to(self, a_crawler, a_position, a_direction):
        a_crawler.change_position_to(a_position)

    # facing

    def turn_crawler_with(self, a_crawler, a_turn):
        a_turn()
